test/per_file_coverage.sh

```diff
@@ -10,6 +10,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/config:96.4"
 "source/common/crypto:88.1"
 "source/common/event:95.1" # Emulated edge events guards don't report LCOV
+"source/common/filesystem/posix:96.5" # FileReadToEndNotReadable keeps failing
```

Comment:
It's a bummer that we need to lower the coverage here :/ I see discussed earlier

> It's possible the test could be conditional (check that the permissions preclude reading the file after they've been changed, and if they don't, just log that the test was skipped and don't do the EXPECT), which would provide the coverage in the coverage environment where the test works, and then 'skip' in the environment where the test doesn't work.

Is that still an option? If so, that sounds preferable. (Every time we lower coverage in 1 directory, we lower the aggregate coverage for the repo which has knock-on effects) But maybe it's not possible?

Reply:
I'm not sure that I quite understand your comment. Have we tried to make the test conditional? If so, did that work? If not, why did it not work? Maybe you've already discussed this and I didn't see it.

Reply:
yes we did discuss that as the thread has grown long for this because for coverage the default failed threshold is just amended to 96.5 upperbound for smaller fix like this. Per @ravenblackx and @phlax discussion lets keep the current PR as suggested. Any future amendments we do later